mlir/test/lib/Analysis/CMakeLists.txt

```diff
@@ -21,12 +21,18 @@ add_mlir_library(MLIRTestAnalysis
   EXCLUDE_FROM_LIBMLIR
 
   LINK_LIBS PUBLIC
+  MLIRTestDialect
+  )
+
+# Since this library is excluded from libMLIR, link it to the dylib
+# to ensure that it can be used in flang without implicitly pulling in
+# static libraries.
```

Comment:
Don't think we need this comment (or at least it's weird to have it in some places but not others).

Reply:
Removed them and added a short note to `EXCLUDE_FROM_LIBMLIR` doc.